classes/class.tapestry-user-progress.php

```diff
@@ -218,12 +197,17 @@ private function _complete()
         update_user_meta($this->_userId, 'tapestry_'.$this->postId.'_node_completed_'.$this->nodeMetaId, true);
     }
 
-    private function _completeQuestion($questionId)
+    private function _completeQuestion($questionId, $answerData, $answerType)
     {
-        $nodeMetadata = get_metadata_by_mid('post', $this->nodeMetaId)->meta_value;
-        $quiz = $this->_getQuizProgress($this->nodeMetaId, $nodeMetadata, $this->_userId);
-        $quiz[$questionId]['completed'] = true;
-        update_user_meta($this->_userId, 'tapestry_'.$this->postId.'_node_quiz_'.$this->nodeMetaId, $quiz);
+        $userAnswer = get_user_meta($this->_userId, 'tapestry_'.$this->postId.'_node_quiz_'.$this->nodeMetaId.'_question_'.$questionId, true);
+        if ($userAnswer === "") {
+          $newUserAnswer[$answerType] = $answerData;
```

Comment:
I followed the steps here on creating php associative arrays. I don't think an error will occur here.
<img width="913" alt="Screen Shot 2021-06-16 at 4 15 31 PM" src="https://user-images.githubusercontent.com/30706624/122306894-21234600-cebe-11eb-92df-5b72226758e2.png">

Reply:
That's interesting. I didn't know you could do that. But I did some reading into this and the [official php documentation](https://www.php.net/manual/en/language.types.array.php) says that this way of initializing an array is discouraged as it could have some unwanted consequences. You can actually avoid your whole if/else statement if you put `|| array()` at the end of line 146 so that if it's an empty string, [it evaluates to Boolean false](https://www.php.net/manual/en/language.types.boolean.php) and so would return `array()` as the value of `$userAnswer` and you can just keep the content of your else statement that way.